scrapy/crawler.py

```diff
@@ -151,8 +151,7 @@ def crawl(self, *args: Any, **kwargs: Any) -> Generator[Deferred[Any], Any, None
             self._apply_settings()
             self._update_root_log_handler()
             self.engine = self._create_engine()
-            start_requests = iter(self.spider.start_requests())
-            yield self.engine.open_spider(self.spider, start_requests)
+            yield self.engine.open_spider(self.spider)
```

Comment:
There is a small difference in case start_requests is not a generator function (when it's a regular function which returns a list). But I don't think there should be any practical implications, it seems we can ignore this.

Reply:
https://github.com/scrapy/scrapy/pull/6729/commits/97a0fed020926af464631d31bdfb354a12aea8c3